vmm/src/config.rs

```diff
@@ -2165,6 +2174,15 @@ impl VmConfig {
             }
         }
 
+        if let Some(balloon) = &self.balloon {
+            if balloon.size >= self.memory.size {
```

Comment:
That's partially wrong as you could have memory size being 0 if there are memory zones along with it. Here you're checking compared to the `size` provided through the `size` parameter from `--memory` option. But in case of memory zones, we should get the total RAM from the addition of all the zones.
That's why this check should be performed from the `MemoryManager` or `DeviceManager` rather than from the config. And if you're thinking about doing the addition of the memory zones from config.rs, that's also not possible as you would perform the check only for the CLI use case.

Reply:
MemoryManager has no concept about the balloon. Instead I just added the zone sizes to the RAM size which should handle this case.

Reply:
You're not comparing against your variable `ram_size` that you just introduced.

But be aware that you're only doing this check for the CLI case. Someone using the HTTP API won't get any error if he tries to create a balloon bigger than the RAM.

Reply:
> You're not comparing against your variable `ram_size` that you just introduced.
> 

Oops.

> But be aware that you're only doing this check for the CLI case. Someone using the HTTP API won't get any error if he tries to create a balloon bigger than the RAM.

The whole point of `::validate()` and why it was introduced was to do common checking for both the CLI and the `vm.create` API end point.

Reply:
My bad, I completely forgot it was invoked from `Vm::new_from_memory_manager()`.